cmd/otel-allocator/watcher/file.go

```diff
@@ -37,6 +39,8 @@ func (f *FileWatcher) Start(upstreamEvents chan Event, upstreamErrors chan error
 	go func() {
 		for {
 			select {
+			case _ = <-f.closer:
```

Comment:
This fixes a bug found that was causing the TA to panic on a configmap change

Reply:
Normally, I would agree but in order to actually run this new code which had configuration changes (something a user would most likely do when enabling this feature) this change was required to ensure the TA wouldn't panic. I can split it out if we think that's worth it, but I kept it in so that users could progressively upgrade without panics.